File: how-many-numbers-are-smaller-than-the-current-number.kt
class Solution {
    fun smallerNumbersThanCurrent(nums: IntArray): IntArray {
        val numList: MutableList<Int> = mutableListOf()
        for (outerNumber in nums) {
          var smallerCount = 0
          for (innerNumber in nums) {
            if (innerNumber < outerNumber) smallerCount += 1
          }
          numList.add(smallerCount)
        }
        return numList.toIntArray()
    }
}

File: how-many-numbers-are-smaller-than-the-current-number.rs
impl Solution {
    pub fn smaller_numbers_than_current(nums: Vec<i32>) -> Vec<i32> {
        let mut v = vec![];
        for i in 0..nums.len() {
            let mut sum = 0;
            for j in 0..nums.len() {
                if nums[j] < nums[i] {
                    sum += 1;
                }
            }
            v.push(sum);
        }
        return v;
    }
}
